libyul/YulStack.cpp

```diff
@@ -164,14 +166,39 @@ void YulStack::optimize(Object& _object, bool _isCreation)
 	if (EVMDialect const* evmDialect = dynamic_cast<EVMDialect const*>(&dialect))
 		meter = std::make_unique<GasMeter>(*evmDialect, _isCreation, m_optimiserSettings.expectedExecutionsPerDeployment);
 
+	auto [optimizeStackAllocation, yulOptimiserSteps, yulOptimiserCleanupSteps] = [&]() -> std::tuple<bool, std::string, std::string>
+	{
+		if (!m_optimiserSettings.runYulOptimiser)
+		{
+			// Yul optimizer disabled, but empty sequence (:) explicitly provided
+			if (OptimiserSuite::isEmptyOptimizerSequence(m_optimiserSettings.yulOptimiserSteps + ":" + m_optimiserSettings.yulOptimiserCleanupSteps))
```

Comment:
By the way, we won't change this now, but in retrospect I think that allowing weird whitespace in the sequence in the first place was a bad idea. I mean, it's fine to allow user to do that, but we should have been just removing all the fluff once, at the entry point in the CLI/`StandardCompiler` and everywhere else expecting it to have no weird quirks.

Reply:
We'd still have to preserve the user input for metadata, no? Or would stripping the sequence of whitespace before validation be fine? In any case, I agree that newlines and whitespace are a hassle, and we'd be better off without.

Reply:
Well, I don't think we really preserve settings exactly in metadata in all cases. The `optimiserSettings` struct does not map 1:1 to settings available to the user, so I suspect there are cases were we can't map back unambiguously. We'd have to discuss this, but I think that stripping stuff that does not affect behavior would be okay.

Reply:
I'm not sure why we'd need to discuss this :-) - it's definitely better to remove the whitespace and not have it in metadata. Ideally metadata should be as normalized as possible (so settings guaranteed to produce equal results should yield equal metadata when possible), this is rather the opposite. But alright, indeed no need to mess with it here.